src/main/java/tech/jhipster/lite/module/infrastructure/secondary/javadependency/gradle/GradleCommandHandler.java

```diff
@@ -269,4 +310,60 @@ private void addPluginConfiguration(GradlePluginConfiguration pluginConfiguratio
       ContentReplacers.of(new MandatoryFileReplacer(new JHipsterProjectFilePath(BUILD_GRADLE_FILE), replacer))
     );
   }
+
+  private void enablePrecompiledScriptPlugins() {
+    addFileToProject(from("buildtool/gradle/buildSrc/build.gradle.kts.template"), to("buildSrc/build.gradle.kts"));
+  }
+
+  private File scriptPluginForProfile(BuildProfileId buildProfileId) {
+    return projectFolder.filePath("buildSrc/src/main/kotlin/profile-%s.gradle.kts".formatted(buildProfileId.value())).toFile();
+  }
+
+  private void addProfileActivation(AddJavaBuildProfile command) {
+    MandatoryReplacer replacer = new MandatoryReplacer(
+      new RegexNeedleBeforeReplacer(always(), GRADLE_PROFILE_ACTIVATION_NEEDLE),
+      fillProfileActivationTemplate(command)
+    );
+    fileReplacer.handle(
+      projectFolder,
+      ContentReplacers.of(new MandatoryFileReplacer(new JHipsterProjectFilePath(BUILD_GRADLE_FILE), replacer))
+    );
+  }
+
+  private static String fillProfileActivationTemplate(AddJavaBuildProfile command) {
+    Optional<Boolean> isActiveByDefault = command.activation().flatMap(BuildProfileActivation::activeByDefault);
+
+    return (isActiveByDefault.orElse(false) ? PROFILE_DEFAULT_ACTIVATION_CONDITIONAL_TEMPLATE : PROFILE_CONDITIONAL_TEMPLATE).formatted(
+        command.buildProfileId(),
+        command.buildProfileId()
+      );
+  }
+
+  private void addScriptPluginForProfile(BuildProfileId buildProfileId) {
+    addFileToProject(
+      from("buildtool/gradle/buildSrc/src/main/kotlin/profile.gradle.kts.template"),
+      to("buildSrc/src/main/kotlin/profile-%s.gradle.kts".formatted(buildProfileId))
+    );
+  }
+
+  private void addFileToProject(JHipsterSource source, JHipsterDestination destination) {
+    if (!projectFolder.fileExists(destination.get())) {
```

Comment:
You should inverse logic to do an early return.
```suggestion
    if (projectFolder.fileExists(destination.get())) {
      return;
    }
```